integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -29,14 +29,26 @@
 // fraction too long.
 
 import { expect } from "chai";
-import { BSON, ClientResetMode, FlexibleSyncConfiguration, Realm, SessionStopPolicy } from "realm";
+import {
+  BSON,
+  ClientResetMode,
+  ConfigurationWithSync,
+  FlexibleSyncConfiguration,
+  Realm,
+  SessionStopPolicy,
+  SyncError,
+} from "realm";
 
 import { authenticateUserBefore, importAppBefore, openRealmBeforeEach } from "../../hooks";
-import { DogSchema, IPerson, PersonSchema } from "../../schemas/person-and-dog-with-object-ids";
+import { Dog, IPerson, Person, PersonSchema } from "../../schemas/person-and-dog-with-object-ids";
 import { closeAndReopenRealm, closeRealm } from "../../utils/close-realm";
 import { expectClientResetError } from "../../utils/expect-sync-error";
 
-const FlexiblePersonSchema = { ...PersonSchema, properties: { ...PersonSchema.properties, nonQueryable: "string?" } };
+class FlexiblePerson extends Person {
```

Comment:
I figured it'd be nice to just use class-based schemas in these tests. Not sure if I should just not extend Person and move all the information here, since this seems to be our preference generally.

Reply:
🚀